api/controllers/login.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const https = require('https')
+const jwt = require('jsonwebtoken')
+const logger = require('../../config/logger')
+const util = require('../util/util')
+
+/* istanbul ignore next */
+const get = (request, response) => {
+  request.body = {
+    ticket: request.query['ticket']
+  }
+
+  post(request, response)
+}
+
+const post = (request, response) => {
+  var serviceTicket = request.body['ticket']
+
+  if (!serviceTicket) {
+    util.buildUnauthorizedResponse(response)
+    return
+  }
+
+  validateTicket(serviceTicket, (data) => {
+    if (!data) {
+      util.buildInternalErrorResponse(response)
+      return
+    }
+
+    var json = JSON.parse(data)
+    if (json.serviceResponse.authenticationFailure) {
+      logger.debug('Login failure: ' + json.serviceResponse.authenticationFailure.description)
+      util.buildUnauthorizedResponse(response)
+      return
+    }
+    var guid = json.serviceResponse.authenticationSuccess.attributes.theKeyGuid[0].toLowerCase()
+    buildJwt(guid, response)
+  })
+}
+
+const validateTicket = (serviceTicket, callback) => {
+  var path = '/cas/p3/serviceValidate'
```

Comment:
yeah, if we ever determine we need to auth via proxy tickets with this API, we will have to change this to `proxyValidate`, but that can wait until the need arises.

Reply:
my preference on configuring the CAS server itself has always been to use "https://thekey.me/cas" as the configured CAS server base URL, then define all endpoint paths based on that. (e.g. /p3/serviceValidate)

But from the looks of how you generate the http request, configuring the app that way would cause more hassle than it would be worth.